native/src/components/MapView.tsx

```diff
@@ -94,31 +109,46 @@ const MapView = ({ boundingBox, featureCollection, currentFeature }: MapViewProp
       ? 'location-searching'
       : 'location-disabled'
 
-  const onPress = useCallback(async (pressedLocation: Feature) => {
-    if (!mapRef?.current || !cameraRef?.current || !pressedLocation.properties) {
-      return
-    }
-    const featureCollection = await mapRef.current.queryRenderedFeaturesAtPoint(
-      [pressedLocation.properties.screenPointX, pressedLocation.properties.screenPointY],
-      undefined,
-      [featureLayerId]
-    )
-    const feature = featureCollection?.features?.find((it): it is Feature<Point> => it.geometry.type === 'Point')
-    if (feature) {
-      setSelectedFeature(feature)
-      const {
-        geometry: { coordinates }
-      } = feature
-      cameraRef.current.flyTo(coordinates)
-    } else {
-      setSelectedFeature(null)
-    }
-  }, [])
+  // set the user location coordinate once to calculate distance for all pois
+  const onLocationUpdate = (location: MapboxGL.Location): void => {
+    !userLocation && setUserLocation([location.coords.longitude, location.coords.latitude])
+  }
```

Comment:
Perhaps it would be better to only update the user location if the change in distance is smaller than `0.1` kilometers.

Reply:
it only happens once at the moment so if there is no userLocation. There was no requirement that we should keep it up to date if the position changes, since we are not a navigation app ;)